pkg/util/util.go

```diff
@@ -98,6 +102,25 @@ func SecretData(client kubernetes.Interface, namespace string, ks corev1.SecretK
 	return data, nil
 }
 
+// GetDataFromSecret returns secret data value for a given secret/key combination or error if secret or key is not found
+func GetDataFromSecret(client client.Client, namespace string, ks corev1.SecretKeySelector) ([]byte, error) {
```

Comment:
Thanks for thorough write-up!

Well, `GetDataFromSecret` was just intended to replace the `SecretData` for the sake of consistency. So, I hadn't given much thought about whether we really still need it in the first place.

I see the value of such functions when they provide extra options and/or project-specific functionality, such as including some conversion logic from a higher level struct to lower level one/s or one can add a label selector for a label that is assumed to exist in all crossplane-managed resources, so that secret selection happens only for these resources. But I don't see us having a plan/need for this at least for now. So, not much value really.

I think we can close this and I can change the `ProviderCredentials` to use `client.Client` instead of `SecretData` and move on with the conflict fix.